app/src/main/java/org/xbmc/kore/ui/sections/file/MediaFileListFragment.java

```diff
@@ -717,9 +717,17 @@ public static FileLocation newInstanceFromItemFile(Context context, ListType.Ite
                     artUrl = itemFile.thumbnail;
                     break;
                 case ListType.ItemBase.TYPE_ALBUM:
+                    title = itemFile.displayartist + " | " + itemFile.album;
+                    details = new File(itemFile.file).getName();
```

Comment:
I don't particularly like that we are creating a `File` object for each item on the directory. I get that it is the easiest way to get the filename all parsed up, but it is kind of wasteful.
Furthermore there's a probable bug here: take a look at the `getName` implementation and you'll notice that it used a `fileSystemSeparator` to get the last part of the name. The thing is, this always returns Android's file system separator (ext4 `\`), not the one where Kodi resides, so if Kodi is on windows, the path probably contains `/` and this won't work. Haven't tested this tough, it's just an hypothesis.